### PR TITLE
migrated to new resource interface

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2056.

### CODE REVIEW

File: service/resource/configmap/resource.go [hunk 1]
```diff
@@ -193,39 +223,18 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
...
-	{
-		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out which config maps have to be deleted")
-
-		for _, currentConfigMap := range currentConfigMaps {
-			if !containsConfigMap(desiredConfigMaps, currentConfigMap) {
-				configMapsToDelete = append(configMapsToDelete, currentConfigMap)
-			}
-		}
-
-		r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d config maps that have to be deleted", len(configMapsToDelete)))
```
Comment (xh3b4sd) [inline service/resource/configmap/resource.go]: When I implemented this I considered deletion of resources during updates. I guess this was never really useful. So it should be ok to get rid of it and implement it in case we need it. This was part of the learning curve on my side.

File: service/resource/configmap/resource.go [hunk 2]
```diff
@@ -252,19 +261,19 @@ func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desire
...
-	return configMapsToCreate, configMapsToDelete, configMapsToUpdate, nil
+	return configMapsToUpdate, nil
 }
 
 func (r *Resource) Name() string {
 	return Name
 }
 
-func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState interface{}) error {
+func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
```
Comment (xh3b4sd) [inline service/resource/configmap/resource.go]: I would like to move to separate files in a separate PR. I did not do it here to keep the diff as clean as possible.

File: service/resource/configmap/resource_test.go [hunk 3]
```diff
@@ -697,93 +695,12 @@ func Test_Resource_CloudConfig_GetUpdateState(t *testing.T) {
...
 			CurrentState:                         []*apiv1.ConfigMap{},
 			DesiredState:                         []*apiv1.ConfigMap{},
-			ExpectedConfigMapsToCreate:           nil,
-			ExpectedConfigMapsToDelete:           nil,
-			ExpectedConfigMapsToUpdate:           nil,
-			ExpectedMessageContextConfigMapNames: nil,
-		},
-
-		// Test 2, in case current state and desired state are equal the create,
-		// delete and update state should be empty.
```
Comment (xh3b4sd) [inline service/resource/configmap/resource_test.go]: Some cases for the configmap and deployment tests are removed because we removed the support of resource deletion during updates.

File: service/service.go [hunk 4]
```diff
@@ -13,11 +13,11 @@ import (
...
 	"github.com/giantswarm/micrologger"
-	"github.com/giantswarm/operatorkit/client/k8s"
+	"github.com/giantswarm/operatorkit/client/k8sclient"
 	"github.com/giantswarm/operatorkit/framework"
-	"github.com/giantswarm/operatorkit/framework/logresource"
-	"github.com/giantswarm/operatorkit/framework/metricsresource"
-	"github.com/giantswarm/operatorkit/framework/retryresource"
+	"github.com/giantswarm/operatorkit/framework/resource/logresource"
+	"github.com/giantswarm/operatorkit/framework/resource/metricsresource"
+	"github.com/giantswarm/operatorkit/framework/resource/retryresource"
```
Comment (xh3b4sd) [inline service/service.go]: More changes towards the vendor update.

**Comment by kopiczko** [comment]
Looks like some tests are gone. This is nice. Thanks!

**Comment by xh3b4sd** [comment]
Successfully tested on lycan. Rolling.
